static/js/stream_create.js

```diff
@@ -307,99 +330,71 @@ export function show_new_stream_modal() {
         $("#announce-new-stream").hide();
     }
     clear_error_display();
-
-    $("#stream-checkboxes label.checkbox").on("change", function (e) {
-        const elem = $(this);
-        const stream_id = Number.parseInt(elem.attr("data-stream-id"), 10);
-        const checked = elem.find("input").prop("checked");
-        const subscriber_ids = new Set(peer_data.get_subscribers(stream_id));
-
-        $("#user-checkboxes label.checkbox").each(function () {
-            const user_elem = $(this);
-            const user_id = Number.parseInt(user_elem.attr("data-user-id"), 10);
-
-            if (subscriber_ids.has(user_id)) {
-                user_elem.find("input").prop("checked", checked);
-            }
-        });
-
-        e.preventDefault();
-    });
 }
 
 export function create_handlers_for_users(container) {
     // container should be $('#people_to_add')...see caller to verify
-    container.on("change", "#user-checkboxes input", update_announce_stream_state);
-
-    // 'Check all' and 'Uncheck all' visible users
-    container.on("click", ".subs_set_all_users", (e) => {
-        $("#user-checkboxes .checkbox").each((idx, li) => {
-            if (li.style.display !== "none") {
-                $(li.firstElementChild).prop("checked", true);
+    function update_checked_state_for_users(value, users) {
+        // Only do full_redraw if we have to render more than 500 users.
+        const full_redraw = !users || users.length > 500;
```

Comment:
How did you pick 500 here?  I would kinda expect that 500 single-row changes would cost more than a full rerender, but could be wrong.  (Or is the UX worse when we do a full rerender, and 500 single-row changes still feels snappy, so that's why you transition there?).

Reply:
And how long did a full rerender take?

Reply:
Based with these numbers, I went with 250 (=> 0.5s worth of incrementals); I don't think it particularly matters where we land as long as it's at least 100 and below 1000.

(Also adjusted the comment to not mention the number 500 and focus on the concept -- that way, we need to churn the codebase less if we change our minds).